airbyte-integrations/connectors/source-blob/source_blob/fileclient.py

```diff
@@ -0,0 +1,263 @@
+
+    def __init__(self, url: str, provider: dict):
+        self._url = url
+        self._provider = provider
+        self._file = None
+        self.logger = AirbyteLogger()
+
+    def __enter__(self):
+        return self._file
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    @property
+    @abstractmethod
+    def url(self) -> str:
+        """Convert URL to remove the URL prefix (scheme)
+        :return: the corresponding URL without URL prefix / scheme
+        """
+
+    @property
+    def fullurl(self):
+        return f"{self.storage_scheme}{self.url}"
+
+    @property
+    @abstractmethod
+    def storage_scheme(self) -> str:
+        """Convert Storage Names to the proper URL Prefix
+        :return: the corresponding URL prefix / scheme
+        """
+        # storage_name = self._provider["storage"].upper()
+        # parse_result = urlparse(self._url)
+        # if storage_name == "HTTPS":
+        #     return "https://"
+        # elif storage_name == "SSH" or storage_name == "SCP":
+        #     return "scp://"
+        # elif storage_name == "SFTP":
+        #     return "sftp://"
+        # elif storage_name == "WEBHDFS":
+        #     return "webhdfs://"
+        # elif storage_name == "LOCAL":
+        #     return "file://"
+        # elif parse_result.scheme:
+        #     return parse_result.scheme
+        # self.logger.error(f"Unknown Storage provider in: {self.fullurl}")
+        # return ""
+
+    @property
+    @abstractmethod
+    def last_modified(self) -> datetime:
+        """ TODO Docstring """
+
+    def close(self):
+        if self._file:
+            self._file.close()
+            self._file = None
+
+    def open(self, binary=False):
+        self.close()
+        self._file = self._open(binary=binary)
+        return self
+
+    @abstractmethod
+    def _open(self, binary):
+        """TODO docstring"""
+        # mode = "rb" if binary else "r"
+        # storage = self.storage_scheme
+        # url = self._url
+        # if storage == "webhdfs://":
+        #     host = self._provider["host"]
+        #     port = self._provider["port"]
+        #     return smart_open.open(f"webhdfs://{host}:{port}/{url}", mode=mode)
+        # elif storage in ("ssh://", "scp://", "sftp://"):
+        #     user = self._provider["user"]
+        #     host = self._provider["host"]
+        #     port = self._provider.get("port", 22)
+        #     # Explicitly turn off ssh keys stored in ~/.ssh
+        #     transport_params = {"connect_kwargs": {"look_for_keys": False}}
+        #     if "password" in self._provider:
+        #         password = self._provider["password"]
+        #         uri = f"{storage}{user}:{password}@{host}:{port}/{url}"
+        #     else:
+        #         uri = f"{storage}{user}@{host}:{port}/{url}"
+        #     return smart_open.open(uri, transport_params=transport_params, mode=mode)
+        # return smart_open.open(self.fullurl, mode=mode)
```

Comment:
Copied from existing file connector, left in commented out for reference but will most likely remove

Reply:
what will you replace it with?

Reply:
this abstract method gets overridden by child classes (e.g. FileClientS3), my thinking in leaving this there for now was that when we come to implement FileClientSsh etc. we have this reference retained from the files source.